### ISSUE TITLE
memory growth with libtorch backend in Juce plugin example

### ISSUE THREAD

**Issue by cccstbccc** [tu-studio/anira#45]
when using the libtorch backend with MODEL_TO_USE 6 the standalone Juce example shows quite heavy memory growth, using a:

`torch::NoGradGuard no_grad_guard;`

in InferenceThread::inference as outlined in the attached patch seems to solve the problem:

```
void InferenceThread::inference(std::shared_ptr<SessionElement> session, std::vector<BufferF>& input, std::vector<BufferF>& output) {
#ifdef USE_LIBTORCH
    if (session->m_current_backend.load(std::memory_order_relaxed) == LIBTORCH) {
        torch::NoGradGuard no_grad_guard;
        if (session->m_libtorch_processor != nullptr) {
            session->m_libtorch_processor->process(input, output, session);
        }
        else {
            session->m_default_processor.process(input, output, session);
            LOG_ERROR << "[ERROR] LibTorch model has not been provided. Using default processor." << std::endl;
        }
    }
#endif
```

i have not enough insight, yet, to know if this is the correct solution though, so i did not create a merge request

[anira-12-31-23.patch](https://github.com/user-attachments/files/22491779/anira-12-31-23.patch)

**Comment by vackva** [comment]
Hi Stefan, thanks for the bug report! We’ll take a look and get back to you soon

**Comment by faressc** [comment]
Hi there,

Thanks for pointing this out! I managed to reproduce both the extensive memory growth and the fix. I agree that this needs to be addressed urgently. As this only occurs with the RAVE models, the issue may be that their model export is missing a 'no_grad' option or something similar. To counteract this problem in other models as well, I think it's reasonable for the LibTorchProcessor to disable gradients completely.

I have also tried creating our input tensors with: torch::TensorOptions options = torch::TensorOptions().requires_grad(false); or torch::from_blob(m_input_data[i].data(), m_inference_config.get_tensor_input_shape(anira::InferenceBackend::LIBTORCH)[i]).set_requires_grad(false). Neither of these solved the memory growth issue for the RAVE model.

I have pushed a commit to the 2.0.3 branch.

Cheers,
Fares
